src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs

```diff
@@ -672,31 +667,34 @@ private static void AddDependencies(IEnumerable<LibraryDependency> dependencies,
         /// <summary>
         /// Convert from the expected nupkg path to the on disk path.
         /// </summary>
-        private static IEnumerable<LockFileItem> ConvertToProjectPaths(
+        private static List<LockFileItem> ConvertToProjectPaths(
             Dictionary<string, ProjectRestoreMetadataFile> fileLookup,
             string projectDir,
-            IEnumerable<LockFileItem> items)
+            IList<LockFileItem> items)
         {
-            foreach (var item in items)
+            var results = new List<LockFileItem>(items.Count);
+            foreach (var item in items.NoAllocEnumerate())
```

Comment:
A `for` loop would have (margionally) less overhead.

Reply:
Yes it will :D 
I'm actually refactoring this since it's unnecessarily heavy so this might become irrelevant.